commpy/channelcoding/README.md

```diff
@@ -39,10 +39,130 @@ Redundancy of the channel coding schemes influences (decreases) bit rate. Actual
 To change the code rate (k/n) of the block code dimensions of the Generator matrix can be changed:
 ![blockcoderate](https://raw.githubusercontent.com/kirlf/CSP/master/FEC/assets/coderateblock.png)
 
-To change the coderate of the continuous code, e.g. [convolutional code](https://github.com/kirlf/CSP/blob/master/FEC/Convolutional%20codes%20intro.md), **puncturing** procedure is frequently used:
+To change the coderate of the continuous code, e.g. convolutional code, **puncturing** procedure is frequently used:
 
 ![punct](https://raw.githubusercontent.com/kirlf/CSP/master/FEC/assets/punct.png)
 
+## Example
+
+Let us consider implematation of the **convolutional codes** as an example:
+
+<img src="https://habrastorage.org/webt/v3/v5/w2/v3v5w2gbwk34nzk_2qt25baoebq.png" width="500"/>
+
+*Main modeling routines: random message genaration, channel encoding, baseband modulation, additive noise (e.g. AWGN), baseband demodulation, channel decoding, BER calculation.*
+
+```python
+import numpy as np
+import commpy.channelcoding.convcode as cc
+import commpy.modulation as modulation
+
+def BER_calc(a, b):
+    num_ber = np.sum(np.abs(a - b))
+    ber = np.mean(np.abs(a - b))
+    return int(num_ber), ber
+
+N = 100000 #number of symbols per the frame
+message_bits = np.random.randint(0, 2, N) # message
+
+M = 4 # modulation order (QPSK)
+k = np.log2(M) #number of bit per modulation symbol
+modem = modulation.PSKModem(M) # M-PSK modem initialization 
+```
+
+The [following](https://en.wikipedia.org/wiki/File:Conv_code_177_133.png) convolutional code will be used:
+
+![](https://upload.wikimedia.org/wikipedia/commons/thumb/b/b3/Conv_code_177_133.png/800px-Conv_code_177_133.png)
+
+*Shift-register for the (7, [171, 133]) convolutional code polynomial.*
+
+Convolutional encoder parameters:
+
+```python
+rate = 1/2 # code rate
+L = 7 # constraint length
+m = np.array([L-1]) # number of delay elements
+generator_matrix = np.array([[0o171, 0o133]]) # generator branches
```

Comment:
`generator_matrix` use a decimal representation rather than a octal one (see #16).

Reply:
Thanks for the reference!
Does it mean if required to implement something like the following MATLAB expression:

``` matlab
convEncoder = comm.ConvolutionalEncoder(poly2trellis(7, [171 133]))
``` 

I should use:  

``` python
generator_matrix = np.array([[121, 91]])
```

If yes, this completely matches with:

``` python
generator_matrix = np.array([[0o171, 0o133]])
```
Isn't it? 

Thank you for your attention!

Reply:
My bad. You are right. Leave it like this :)

Reply:
Ok :)